grails-app/services/org/bbop/apollo/FeatureService.groovy

```diff
@@ -2468,4 +2472,339 @@ public void setTranslationEnd(Transcript transcript, int translationEnd) {
+
+        if (currentFeatureJsonObject.has(FeatureStringEnum.PARENT_TYPE.value)) {
+            currentFeatureJsonObject.get(FeatureStringEnum.PARENT_TYPE.value).name = topLevelFeatureType
+        }
+        currentFeatureJsonObject.get(FeatureStringEnum.TYPE.value).name = type
+        currentFeatureJsonObject.put(FeatureStringEnum.USERNAME.value, currentFeatureJsonObject.get(FeatureStringEnum.OWNER.value.toLowerCase()))
+        currentFeatureJsonObject.remove(FeatureStringEnum.PARENT_ID.value)
+        currentFeatureJsonObject.remove(FeatureStringEnum.ID.value)
+        currentFeatureJsonObject.remove(FeatureStringEnum.OWNER.value.toLowerCase())
+        currentFeatureJsonObject.remove(FeatureStringEnum.DATE_CREATION.value)
+        currentFeatureJsonObject.remove(FeatureStringEnum.DATE_LAST_MODIFIED.value)
+        if (currentFeatureJsonObject.has(FeatureStringEnum.CHILDREN.value)) {
+            for(JSONObject childFeature : currentFeatureJsonObject.get(FeatureStringEnum.CHILDREN.value)) {
+                childFeature.remove(FeatureStringEnum.ID.value)
+                childFeature.remove(FeatureStringEnum.OWNER.value.toLowerCase())
+                childFeature.remove(FeatureStringEnum.DATE_CREATION.value)
+                childFeature.remove(FeatureStringEnum.DATE_LAST_MODIFIED.value)
+                childFeature.get(FeatureStringEnum.PARENT_TYPE.value).name = type
+            }
+        }
+
+
+        if (!singletonFeatureTypes.contains(originalType) && rnaFeatureTypes.contains(type)) {
+            // *RNA to *RNA
+            if (transcriptList.size() == 1) {
+                featureRelationshipService.deleteFeatureAndChildren(parentGene)
+            }
+            else {
+                featureRelationshipService.removeFeatureRelationship(parentGene, feature)
+                featureRelationshipService.deleteFeatureAndChildren(feature)
+            }
+
+            log.debug "Converting ${originalType} to ${type}"
+
+            Transcript transcript = null
+            if (type == MRNA.alternateCvTerm) {
+                // *RNA to mRNA
+                transcript = generateTranscript(currentFeatureJsonObject, sequence, true)
+                setLongestORF(transcript)
+            }
+            else {
+                // *RNA to *RNA
+                transcript = addFeature(currentFeatureJsonObject, sequence, user, true)
+                setLongestORF(transcript)
+            }
+
+            Gene newGene = transcriptService.getGene(transcript)
+            newGene.symbol = parentGeneSymbol
+            newGene.description = parentGeneDescription
+
+            parentGeneDbxrefs.each { it ->
+                DBXref dbxref = new DBXref(
+                        db: it.db,
+                        accession: it.accession,
+                        version: it.version,
+                        description: it.description
+                ).save(flush: true)
```

Comment:
does it work without the flush here?   If you are doing a flush save at the end, it will flush all of the upstream saves as well